recipes/tinyspline/all/patches/fix-cmake.patch

```diff
@@ -0,0 +1,124 @@
+--- a/CMakeLists.txt
```

Comment:
@SpaceIm 
FYI, I'm in the process of creating a pr/patch for upstream that addresses the changes in this patch.

Reply:
@SpaceIm 
https://github.com/msteinbeck/tinyspline/pull/167

Can you please review?
And also update the patch so it is future proof?

Reply:
Two comments:
- I would use generate_export_headers instead of handwritten definitions
- add ALIAS targets `tinyspline::tinyspline` and `tinysplinecxx::tinysplinecxx`

Side note: this lib creates on config file per target... can't be modeled in conan

Reply:
- I opted for not using `generate_export_header` to keep the the install simple (=2 headers).
I'll ask the repo's owner opinion if/when he gets to reviewing the pr.
- you're correct about the namespaces, I'll ask him about what names he prefers.

Reply:
@SpaceIm 
My pr just got merged.

I think the main required changes are:
- add the following options, and initialize to `OFF`: TINYSPLINE_BUILD_EXAMPLES, TINYSPLINE_BUILD_TESTS, TINYSPLINE_BUILD_DOCS, TINYSPLINE_WARNINGS_AS_ERRORS
- create the correct package_info files+targets (upstream was fine with tinyspline::tinyspline + tinysplinecxx::tinysplinecxx) so we can't currently create them 100%

I think your patch can remain, or you can use the new version tag and ditch the patch.

Reply:
Would it be fine for them to have a new release with my changes?

Reply:
I can backport your patch in this recipe, or just wait for a new release

Reply:
There will probably be a new release soon™️

Reply:
Hi @SpaceIm,

I will prepare a new release as soon as possible. I still need to make some changes but they won't affect your recipe.